src/extension/features/budget/show-available-after-savings/index.js

```diff
@@ -0,0 +1,65 @@
+import { Feature } from 'toolkit/extension/features/feature';
+import { isCurrentRouteBudgetPage } from 'toolkit/extension/utils/ynab';
+import { getEmberView } from 'toolkit/extension/utils/ember';
+import { addToolkitEmberHook } from 'toolkit/extension/utils/toolkit';
+import { createBudgetBreakdownElement } from 'toolkit/extension/features/budget/subtract-upcoming-from-available/index';
+
+export class ShowAvailableAfterSavings extends Feature {
+  shouldInvoke() {
+    return isCurrentRouteBudgetPage();
+  }
+
+  invoke() {
+    addToolkitEmberHook(this, 'budget-breakdown', 'didRender', this.run);
```

Comment:
I think you can get rid of your `run` function and just pass `this.showAvailableAfterSavings` it's reasonable to expect we're on the budget page if this is running

Reply:
you'll want to use `this.addToolkitEmberHook` to get the auto-cleanup of the hook and also run your function once on any existing elements

```
invoke() {
  this.addToolkitEmberHook('budget-breakdown', 'didRender', this.run);

  const element = document.querySelector('.budget-breakdown');
  if (element) {
    this.showAvailableAfterSavings(element);
  }
}
```

this is just to make it so if someone loads YNAB with your feature off and then they enable it, your feature actually runs. it's a bit of a paradigm shift from before but adds some cool user value :)